packages/react-dom/src/client/validateDOMNesting.js

```diff
@@ -373,25 +379,56 @@ if (__DEV__) {
       case 'h4':
       case 'h5':
       case 'h6':
-        return ancestorInfo.pTagInButtonScope;
+        return (
```

Comment:
These checks all look very similar, is there a refactoring you can do to make the logic clearer?

Reply:
I've tried putting it into a variable to avoid the repeated code